types/node/vm.d.ts

```diff
@@ -19,16 +20,26 @@ declare module "vm" {
          */
         columnOffset?: number;
     }
-    interface ScriptOptions extends BaseOptions {
-        displayErrors?: boolean;
-        timeout?: number;
-        cachedData?: Buffer;
+
+    interface ScriptOptionsObject extends BaseOptions {
+        cachedData?: NodeJS.TypedArray | DataView;
+        /** @deprecated This option is deprecated in favor of script.createCachedData() */
         produceCachedData?: boolean;
+        /** This option is part of the experimental API for the --experimental-modules flag, and should not be considered stable. */
+        importModuleDynamically?(specifier: string, module: SourceTextModule): Object | SourceTextModule;
     }
+
+    /**
+     * If options is a string, then it specifies the filename.
+     * https://nodejs.org/api/vm.html#vm_constructor_new_vm_script_code_options
+     */
+    type ScriptOptions = ScriptOptionsObject | string;
```

Comment:
I think it would make more sense to leave the interface name `ScriptOptions` and add the `| string` union to the methods that possibly accept strings.

Reply:
Such an approach would be counterintuitive, since when importing "ScriptOptions" type, the developer expects to contain all available configuration options.
I solved this problem as follows:
```typescript
type ScriptOptions = {
        cachedData?: NodeJS.TypedArray | DataView;
        /** @deprecated This option is deprecated in favor of script.createCachedData() */
        produceCachedData?: boolean;
        /** This option is part of the experimental API for the --experimental-modules flag, and should not be considered stable. */
        importModuleDynamically?(specifier: string, module: SourceTextModule): any | SourceTextModule;
    } & BaseOptions | string;
```